src/main/java/brooklyn/entity/proxy/aws/ElbControllerImpl.java

```diff
@@ -173,175 +174,179 @@ protected void startLoadBalancer() {
         String elbName = checkNotNull(checkLoadBalancerName(LOAD_BALANCER_NAME), LOAD_BALANCER_NAME.getName());
 
-        int loadBalancerPort = getRequiredConfig(LOAD_BALANCER_PORT);
-        int instancePort = getRequiredConfig(INSTANCE_PORT);
-        String instanceProtocol = getRequiredConfig(INSTANCE_PROTOCOL);
         String elbScheme = getConfig(LOAD_BALANCER_SCHEME);
-        String loadBalancerProtocol = getRequiredConfig(LOAD_BALANCER_PROTOCOL);
-        Collection<String> securityGroups = getConfig(LOAD_BALANCER_SECURITY_GROUPS);
-        Set<String> subnets = (getConfig(LOAD_BALANCER_SUBNETS) != null) ? ImmutableSet.copyOf(getConfig(LOAD_BALANCER_SUBNETS)) : ImmutableSet.<String>of();
-        String sslCertificateId = getConfig(SSL_CERTIFICATE_ID);
-        Set<String> availabilityZoneNames = getAvailabilityZones(loc);
-        Boolean healthCheckEnabled = getConfig(HEALTH_CHECK_ENABLED);
-        Integer healthCheckInterval = getConfig(HEALTH_CHECK_INTERVAL);
-        Integer healthCheckTimeout = getConfig(HEALTH_CHECK_TIMEOUT);
-        Integer healthCheckHealthyThreshold = getConfig(HEALTH_CHECK_HEALTHY_THRESHOLD);
-        Integer healthCheckUnhealthyThreshold = getConfig(HEALTH_CHECK_UNHEALTHY_THRESHOLD);
 
         LOG.debug("Re-initialising existing ELB: " + elbName);
 
-        LoadBalancerServiceContext loadBalancerServiceContext = null;
         try {
             loadBalancerServiceContext = newLoadBalancerServiceContext(loc.getIdentity(), loc.getCredential());
-            ELBApi api = api(loadBalancerServiceContext);
 
-            LoadBalancer loadBalancer = api.getLoadBalancerApi().get(elbName);
+            LoadBalancer loadBalancer = getELBApi().getLoadBalancerApi().get(elbName);
             Scheme oldScheme = loadBalancer.getScheme().orNull();
             if (oldScheme != null && !oldScheme.value().equals(elbScheme)) {
                 LOG.warn("Existing ELB {} scheme ({}) is different from configuration ({}); cannot modify; continuing",
                         new Object[] {elbName, oldScheme, elbScheme});
             }
 
-            /*
+            configureSecurityGroups(elbName);
+
+            configureAvailabilityZones(elbName, loadBalancer);
+
+            configureSubnets(elbName, loadBalancer);
+
+            configureListeners(elbName, loadBalancer);
+            configureHealthCheck();
+
+            sensors().set(Attributes.HOSTNAME, loadBalancer.getDnsName());
+            }
+        finally {
+            Closeables2.closeQuietly(loadBalancerServiceContext.unwrap());
+        }
+
+    }
+    private void configureSecurityGroups(String elbName){
+        final Collection<String> securityGroups = getConfig(LOAD_BALANCER_SECURITY_GROUPS);
+
+        if (securityGroups != null && securityGroups.size() > 0) {
+            getELBApi().getLoadBalancerApi().applySecurityGroupsToLoadBalancer(elbName, securityGroups != null ? securityGroups : ImmutableList.<String>of());
```

Comment:
Do you need to fetch the existing security groups if some were applied manually?